src/extralit_ocr/jobs.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2024-present, Extralit Labs, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""RQ jobs for PyMuPDF processing with document-specific margin support."""
+
+import asyncio
+import logging
+from typing import Any, Optional, Tuple
+from uuid import UUID
+
+from rq import get_current_job
+
+from ..extract import extract_markdown_with_hierarchy, ExtractionConfig
+
+_LOGGER = logging.getLogger(__name__)
+
+# Try to import database components for margin fetching
+try:

```

Comment:
`extralit-hf-space` always have `extralit_server` as the dependency, so this try-except block should removed.

Btw, for development, can you not use .venv and use a shared python environment for both repos?

Reply:
Thanks. Btw do you know why you can't use 2 separate venv for each repo?

Reply:
yes that's correct. If you have separate venv in the `extralit-hf-space/` path, when you install the python package, only the `extralit_ocr` package is installed in the venv, so it won't have the `extralit_server` package installed.

Reply:
The most recommended approach is to activate only 1 python environment for the entire Extralit development across repos. I personally use `micromamba` since venv tend to only work with the local directory where the code lives.